### PR TITLE
Feature/requesteventuniquely

### PR DESCRIPTION
**Pull request by ScottBTR**
### Description of Change ###

Allow Events to be retrieved by Id.

Note I'm away of the poor exception being thrown in the android version, that has been address in the latest version of the code base.

### Bugs Fixed ###

- Related to issue #

#23 

### API Changes ###

Added: 
 
- `static Task<IEvent> PlatformGetEventByIdAsync(string eventId)`

### Behavioral Changes ###

Instead of one request that pulls back all events with their retrospective data, now pulls back minimum data to build an event list, and then pulls back event data in a separate call by id.

### PR Checklist ###

- [ ] Has tests (if omitted, state reason in description)
- [x] Has samples (if omitted, state reason in description)
- [ ] Rebased on top of master at time of PR
- [ ] Changes adhere to coding standard
- [ ] Updated documentation ([see walkthrough](https://github.com/xamarin/Essentials/wiki/Documenting-your-code-with-mdoc))

### CODE REVIEW

File: Samples/Samples.iOS/Samples.iOS.csproj [hunk 1]
```diff
@@ -115,6 +115,9 @@
     <ErrorReport>prompt</ErrorReport>
     <CodeAnalysisRuleSet>C:\GitHub\xamarin\Essentials\Xamarin.Essentials.ruleset</CodeAnalysisRuleSet>
   </PropertyGroup>
+  <PropertyGroup Condition=" '$(RunConfiguration)' == 'Default' ">
```
Comment (nickrandolph) [inline Samples/Samples.iOS/Samples.iOS.csproj]: We should avoid changing the csproj file unless required. Can you check over why these changes were made and perhaps reverse some of them

File: Samples/Samples/View/CalendarPage.xaml.cs [hunk 2]
```diff
@@ -13,14 +14,17 @@ public CalendarPage()
             InitializeComponent();
         }
 
-        void OnEventTapped(object sender, ItemTappedEventArgs e)
+        async void OnEventTapped(object sender, ItemTappedEventArgs e)
         {
             if (e.Item == null && e.Item is Event)
```
Comment (nickrandolph) [inline Samples/Samples/View/CalendarPage.xaml.cs]: use "e.Item is Event event" to define "event" that can be used below instead of (e.Item as Event)?.Id

File: Xamarin.Essentials/Calendar/Calendar.android.cs [hunk 3]
```diff
@@ -116,10 +149,12 @@ static async Task<IReadOnlyList<IEvent>> PlatformGetEventsAsync(string calendarI
                     HasAttendees = cur.GetInt(eventsProjection.IndexOf(CalendarContract.Events.InterfaceConsts.HasAttendeeData)) == 1,
                     HasExtendedProperties = cur.GetInt(eventsProjection.IndexOf(CalendarContract.Events.InterfaceConsts.HasExtendedProperties)) == 1,
                     Status = cur.GetString(eventsProjection.IndexOf(CalendarContract.Events.InterfaceConsts.Status))
-                });
+                };
+            }
+            else
+            {
+                throw new Exception("o oh");
```
Comment (nickrandolph) [inline Xamarin.Essentials/Calendar/Calendar.android.cs]: Do we want to throw an exception, or just return null if the event isn't found?

File: Xamarin.Essentials/Calendar/Calendar.ios.cs [hunk 4]
```diff
@@ -70,5 +74,22 @@ static async Task<IReadOnlyList<IEvent>> PlatformGetEventsAsync(string calendarI
...
 
         static async Task PlatformRequestCalendarWriteAccess() => await Permissions.RequireAsync(PermissionType.CalendarWrite);
+
+        static async Task<IEvent> PlatformGetEventByIdAsync(string eventId)
+        {
+            await Permissions.RequireAsync(PermissionType.CalendarRead);
+
+            var e = CalendarRequest.Instance.GetCalendarItem(eventId) as EKEvent;
+
+            return new Event
```
Comment (nickrandolph) [inline Xamarin.Essentials/Calendar/Calendar.ios.cs:92]: Null check on e?

File: Samples/Samples/View/CalendarPage.xaml.cs (hunk 2 above)
Comment (ScottBTR) [reply Samples/Samples/View/CalendarPage.xaml.cs]: Altered in new commit

File: Xamarin.Essentials/Calendar/Calendar.android.cs (hunk 3 above)
Comment (ScottBTR) [reply Xamarin.Essentials/Calendar/Calendar.android.cs]: Changed to return NullReferenceException to match the behaviour of iOS (and uwp in a future PR)

File: Xamarin.Essentials/Calendar/Calendar.ios.cs (hunk 4 above)
Comment (ScottBTR) [reply Xamarin.Essentials/Calendar/Calendar.ios.cs:92]: Added in next commit

File: Samples/Samples.iOS/Samples.iOS.csproj (hunk 1 above)
Comment (ScottBTR) [reply Samples/Samples.iOS/Samples.iOS.csproj]: Addressed in next commit, only thing required to keep was global suppression inclusion which seemed to be removed previously by mistake. (it simply covers the 'iOS' case)

**Review by nickrandolph — APPROVED** [review]
